asset/quickstart/search-all-iam-policies/main.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START asset_quickstart_search_all_iam_policies]
+
+// Sample search-all-iam-policies search all iam policies within the given scope.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	asset "cloud.google.com/go/asset/apiv1"
+	"google.golang.org/api/iterator"
+	assetpb "google.golang.org/genproto/googleapis/cloud/asset/v1"
+)
+
+// Command-line flags.
+var (
+	scope = flag.String("scope", "", "Scope of the search.")
+	query = flag.String("query", "", "Query statement.")
+)
+
+func main() {
+	flag.Parse()
+	ctx := context.Background()
+	client, err := asset.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("asset.NewClient: %v", err)
+	}
+
+	pageSize := 0
+	pageToken := ""
+
+	req := &assetpb.SearchAllIamPoliciesRequest{
+		Scope:     *scope,
+		Query:     *query,
+		PageSize:  int32(pageSize),
+		PageToken: pageToken,
```

Comment:
Do we need to include the page size and token if they are the default values (`0` and `""`)?

Reply:
I was thinking listing all the fields in the API request might be easier for users to copy the sample code and make changes on their own.

Reply:
Maybe link to the iterator guidelines in a comment, if you want? We haven't done this for other iterators. So, I was confused when I first saw this -- I thought this was a different type of iterator or something. I think it's reasonable to not show all request options. If people want more, they can look at the definition of the request.

As is, I'm concerned it looks like these are mandatory or somehow used as part of the iteration process.